### ISSUE TITLE
Confussing requirement condition evaluation with aliases and umbrella chart

### ISSUE THREAD

**Issue by groundnuty** [helm/helm#3734]
I've created an *umbrella* chart using 3 aliased copies a *foo* chart (foo1, foo2, foo3). *Foo* chart has a single dependency *foo-feature* that can be enabled or disabled (enabled by default). The picture below illustrates these dependencies. 
![image](https://user-images.githubusercontent.com/241318/37767976-3af74914-2dcc-11e8-8d8e-265e8b4f658f.png)

My goal here is to make a deployment where foo3 has its 'foo-feature' disabled and other aliases have this feature enabled. Desired deployment:

![image](https://user-images.githubusercontent.com/241318/37768226-1de65148-2dcd-11e8-8e80-3354bee587de.png)

The natural course of action it would seem is to edit the values in the umbrella chart:
~~~
foo1:
  foo-feature:
    enabled: true

foo2:
  foo-feature:
    enabled: true

foo3:
  foo-feature:
    enabled: false
~~~

However, regardless of the true/false values I always end up with all 3 *foo-feature* being rendered.

How can I achieve the desired deployment?

I attach a tgz with all 3 charts. My test was to edit the values in the *umbrella* chart and then run:
[example.zip](https://github.com/kubernetes/helm/files/1837304/example.zip)
~~~
helm template umbrella | grep Source: | grep deployment.yaml | grep foo-feature
~~~
to see how many deployments of *foo-feature* were rendered.

~~~
> helm version
Client: &version.Version{SemVer:"v2.8.1", GitCommit:"6af75a8fd72e2aa18a2b278cfe5c7a1c5feca7f2", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.1", GitCommit:"6af75a8fd72e2aa18a2b278cfe5c7a1c5feca7f2", GitTreeState:"clean"}
~~~

**Comment by jascott1** [comment]
@groundnuty Unfortunately this appears to be a consequence of evaluating the condition from the true root of the computed values from the umbrella chart. If you change `foo/requirements.yaml`'s condition to `foo1.foo-feature.enabled` you will get different results (but its not a solution). There are some other issues arising from `alias` and `conditions` not working perfectly together and there is a suggestion to change it to a relative evaluation. 

I fear breaking backwards compatibility but Id like to enable your use case here since its so efficient on disk and facilitates a flat directory structure. If you would be interested in a proposal/PR i would help support it.

**Comment by aelbarkani** [comment]
I have a similar problem. It seems like the values of the 2nd level subchart are not taken into account by helm for rendering.

/kind bug

**Comment by sebastien-prudhomme** [comment]
Any news on this annoying bug?

**Comment by groundnuty** [comment]
I'm still trying to find time to try to write a pull request ;-(

**Comment by derkoe** [comment]
@groundnuty here you go #4783

**Comment by groundnuty** [comment]
@derkoe if you happen to have a binary of helm with those changes somewhere lying around I'm happy to test this against my setup and report back.

**Comment by derkoe** [comment]
I have no access to a Mac - so I could only provide you with Win and Linux binaries. You can easily build your own - grab my [release-2.11](https://github.com/derkoe/helm/tree/release-2.11) branch and build your own (see docs here https://docs.helm.sh/developers/)

**Comment by groundnuty** [comment]
any change to accept this pull request? this feature (aka. bug) has been part of helm for long enough....

**Comment by andrewnazarov** [comment]
For some reason, I doubt this will be merged. A part of disscussion here: https://github.com/helm/helm/issues/4490

**Comment by groundnuty** [comment]
@jascott1 has this issue been resolved in Helm 3.0? I just noticed a release and I'm very eager to port all our charts to 3.0, but I failed to find any mention of this issue in the redesigned dependency system of 3.0.
